platform-service/src/unstract/platform_service/helper.py

```diff
@@ -69,3 +75,84 @@ def get_prompt_instance_from_db(
         cursor.close()
         db_instance.close()
         return data_dict
+
+
+class CostCalculationHelper:
+    def __init__(
+        self,
+        url: str,
+        ttl_days: int,
+        file_path: str,
+        logger: Logger,
+    ):
+        self.ttl_days = ttl_days
+        self.url = url
+        self.file_path = file_path
+        self.logger = logger
+
+    def calculate_cost(self, model_name, input_tokens, output_tokens):
+        cost = 0.0
+        item = None
+        model_prices = self._get_model_prices()
+        if model_prices:
+            item = model_prices.get(model_name)
+        if item:
+            input_cost_per_token = item.get("input_cost_per_token", 0)
+            output_cost_per_token = item.get("output_cost_per_token", 0)
+            cost += input_cost_per_token * input_tokens
+            cost += output_cost_per_token * output_tokens
+        return numpy.format_float_positional(
+            cost,
+            trim="-",
+            precision=10,
+        )
```

Comment:
Isn't this precision a bit too high? :)

Reply:
Yes. But it won't be an issue. Because `trim="-"` removes the tailing zeros if present. There are issues in floating point arithmetic as mentioned [here](https://docs.python.org/2/tutorial/floatingpoint.html). Just added precision of 10 to remove the error in precision. Refer the small snippet below.

![image](https://github.com/Zipstack/unstract/assets/89829542/2c391f8f-5b63-4bf5-a725-5f1f43946e86)

It automatically gives output with precision of '%.17f'.